File: BackEnd/DotNet/WebDigital/WebDigital/Models/TransactionItem.cs
using System;
using System.Collections.Generic;

namespace WebDigital.Models;

public partial class TransactionItem
{
    public int ItemId { get; set; }

    public decimal? Price { get; set; }

    public int? Quantity { get; set; }

    public int? ProductId { get; set; }

    public long? TransactionId { get; set; }

    public virtual Product? Product { get; set; }

    public virtual Transaction? Transaction { get; set; }
}

File: BackEnd/Java/transaction-service/src/main/java/com/digitalshelf/transactionservice/entity/TransactionItem.java
package com.digitalshelf.transactionservice.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;

@Entity
@Table(name = "transaction_item")
@Data
public class TransactionItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "item_id")
    private Integer itemId;

    private BigDecimal price;
    private Integer quantity;

    @Column(name = "product_id")
    private Integer productId;

    @Column(name = "transaction_id")
    private Long transactionId;
}